run-command.el

```diff
@@ -233,10 +236,18 @@ said functions)."
                       command-specs)))
           run-command-recipes))
 
-(defun run-command--ivy-action (selection)
+(defun run-command--ivy-action (selection &optional edit)
   "Execute `SELECTION' from Ivy."
-  (let ((command-spec (cdr selection)))
-    (run-command--run command-spec)))
+  (let* ((command-spec (cdr selection))
+         (command-line (plist-get command-spec :command-line))
+         (final-command-line (if (or edit ivy-current-prefix-arg)
+                                 (read-string "> " (concat command-line " "))
+                               command-line)))
+    (run-command--run (plist-put command-spec
+                                 :command-line final-command-line))))
+
+(defun run-command--ivy-edit-action (selection)
+  (run-command--ivy-action selection t))
```

Comment:
Rather than adding an optional argument to `run-command--ivy-action`, would re-using `ivy-current-prefix-arg` work? Something like:

```emacs-lisp
(defun run-command--ivy-edit-action (selection)
  (let ((invy-current-prefix-arg t))
    (run-command-ivy-action)))
```

Reply:
Oh nice, that's cleaner! Updated.